ShenzhenMod/ShenzhenLocator.cs

```diff
@@ -127,15 +127,12 @@ public static string FindUnpatchedShenzhenExecutable(string shenzhenDir)
             sm_log.InfoFormat("Fetching the latest list of unpatched hashes");
 
             WebRequest webRequest = WebRequest.Create("https://raw.githubusercontent.com/gtw123/ShenzhenMod/master/hashes.json");
-            WebResponse response = webRequest.GetResponse();
+            WebResponse response;
 
-            if (((HttpWebResponse)response).StatusDescription != "OK")
-            {
-                throw new Exception(Invariant($"Cannot fetch hash list. Response: \"{((HttpWebResponse)response).StatusDescription}\""));
-            }
+            try { response = webRequest.GetResponse(); }
+            catch(Exception e) { throw new AggregateException("\n\nAn error occurred fetching the hash list. Please try again.\n\n", e); }
```

Comment:
I think AggregateException is really designed for when you have two separate exceptions, like when you run multiple tasks in parallel and several of them throw exceptions, but you want to report them as one exception altogether. It's probably fine to use it here, but when I run the patcher and it can't find the hashes file, the exception message it prints out seems to contain the nested WebException twice.

So it might be better to just go with a standard `Exception` here. You can still pass the inner (chained) exception into the constructor and both will get printed out fine.

Reply:
Yikes! I'm not sure what I was thinking here. Like I said, I'm a C# noob. 😅